pkg/ingestor/parser/spdx/parse_spdx.go

```diff
@@ -65,10 +77,23 @@ func NewSpdxParser() common.DocumentParser {
 
 func (s *spdxParser) Parse(ctx context.Context, doc *processor.Document) error {
 	s.doc = doc
-	spdxDoc, err := parseSpdxBlob(doc.Blob)
-	if err != nil {
-		return fmt.Errorf("failed to parse SPDX document: %w", err)
+	var spdxDoc *v2_3.Document
+	if doc.Type == processor.DocumentITE6SPDX {
+		var err error
+		itespdx, err := parseITESPDXBlob(doc.Blob)
```

Comment:
This should be part of the processor such the parser does not need to change as it will just receive a valid SPDX document at the end.